lib/connection/connection_config.js

```diff
@@ -483,6 +484,14 @@ function ConnectionConfig(options, validateCredentials, qaMode, clientInfo)
     disableQueryContextCache = options.disableQueryContextCache;
   }
 
+  let loginTimeout = 300;
+  if (Util.exists(options.loginTimeout)) {
+    Errors.checkArgumentValid(Util.isNumber(options.loginTimeout),
+    ErrorCodes.ERR_CONN_CREATE_INVALID_MAX_LOGIN_TIMEOUT);
+
+    loginTimeout = options.loginTimeout > 300 ? options.loginTimeout : 300;
```

Comment:
Maybe max of 300 and loginTimeout

Reply:
As far as I know, the minimum value of this property is 300. It cannot be lower than 300 because of the new strategy and can only increase.

Reply:
Yes, that is right.

Reply:
I've thought about  loginTimeout = Math.max(300, options.loginTimeout);

Reply:
Replaced it.